File: controller.js
import express from "express";
import http from "http";

// For new version
import { Server } from "socket.io";

const app = express();
const server = http.createServer(app);

// For new version
const io = new Server(server);

// Create io with namespace robot
const robotio = io.of("/js");
// Initialize listeners
robotio.on("connection", (socket) => {
  console.log("js connected");

  socket.on("pulse", () => {
    console.log("js Pulse");
  });

  socket.on("disconnect", () => {
    console.log("js disconnected");
  });
});

// Create io with namespace "py"
const cameraio = io.of("/py");
// Initialize listeners
cameraio.on("connection", (socket) => {
  console.log("py connected");

  socket.on("pulse", () => {
    console.log("py pulse");
  });

  socket.on("disconnect", () => {
    console.log("py disconnected");
  });
});

server.listen(3000, () => {
  console.log("listening on *:3000");
});

File: controller.py
import socketio
from aiohttp import web

# Create a new socket.io server and aiohttp web application
sio = socketio.AsyncServer(cors_allowed_origins="*")
app = web.Application()
sio.attach(app)


class JSNamespace(socketio.AsyncNamespace):
    async def on_connect(self, sid, environ):
        print("js connected")

    async def on_pulse(self, sid):
        print("js Pulse")

    async def on_disconnect(self, sid):
        print("js disconnected")


class PYNamespace(socketio.AsyncNamespace):
    async def on_connect(self, sid, environ):
        print("py connected")

    async def on_pulse(self, sid, data):
        print("py pulse")

    async def on_disconnect(self, sid):
        print("py disconnected")


sio.register_namespace(JSNamespace('/js'))
sio.register_namespace(PYNamespace('/py'))

if __name__ == '__main__':
    web.run_app(app, port=3000)
